SU2_CFD/src/solvers/CSpeciesSolver.cpp

```diff
@@ -0,0 +1,529 @@
+  if (config->GetKind_Turb_Model())
+    solver[MESH_0][TURB_SOL]->Postprocessing(geometry[MESH_0], solver[MESH_0], config, MESH_0);
+  // For feature_multicomp this Scalar-Pre only computes the laminar contribution to mass diffusivity
+  solver[MESH_0][SPECIES_SOL]->Preprocessing(geometry[MESH_0], solver[MESH_0], config, MESH_0, NO_RK_ITER,
+                                             RUNTIME_FLOW_SYS, false);
+
+  /*--- Interpolate the solution down to the coarse multigrid levels ---*/
+
+  for (iMesh = 1; iMesh <= config->GetnMGLevels(); iMesh++) {
+    SU2_MPI::Error("Untested code: Remove error and proceed at own risk!", CURRENT_FUNCTION);
+    SU2_OMP_FOR_STAT(omp_chunk_size)
+    for (iPoint = 0; iPoint < geometry[iMesh]->GetnPoint(); iPoint++) {
+      Area_Parent = geometry[iMesh]->nodes->GetVolume(iPoint);
+      su2double Solution_Coarse[MAXNVAR] = {0.0};
+      for (iChildren = 0; iChildren < geometry[iMesh]->nodes->GetnChildren_CV(iPoint); iChildren++) {
+        Point_Fine = geometry[iMesh]->nodes->GetChildren_CV(iPoint, iChildren);
+        Area_Children = geometry[iMesh - 1]->nodes->GetVolume(Point_Fine);
+        Solution_Fine = solver[iMesh - 1][SPECIES_SOL]->GetNodes()->GetSolution(Point_Fine);
+        for (iVar = 0; iVar < nVar; iVar++) {
+          Solution_Coarse[iVar] += Solution_Fine[iVar] * Area_Children / Area_Parent;
+        }
+      }
+      solver[iMesh][SPECIES_SOL]->GetNodes()->SetSolution(iPoint, Solution_Coarse);
+    }
+    END_SU2_OMP_FOR
+
+    solver[iMesh][SPECIES_SOL]->InitiateComms(geometry[iMesh], config, SOLUTION);
+    solver[iMesh][SPECIES_SOL]->CompleteComms(geometry[iMesh], config, SOLUTION);
+
+    /// NOTE TK:: This probably need to be adapted like above
+    solver[iMesh][FLOW_SOL]->Preprocessing(geometry[iMesh], solver[iMesh], config, iMesh, NO_RK_ITER, RUNTIME_FLOW_SYS,
+                                           false);
+    solver[iMesh][SPECIES_SOL]->Postprocessing(geometry[iMesh], solver[iMesh], config, iMesh);
+  }
+
+  /*--- Go back to single threaded execution. ---*/
+  SU2_OMP_MASTER {
+    /*--- Delete the class memory that is used to load the restart. ---*/
+
+    delete[] Restart_Vars;
+    Restart_Vars = nullptr;
+    delete[] Restart_Data;
+    Restart_Data = nullptr;
+  }
+  END_SU2_OMP_MASTER
+  SU2_OMP_BARRIER
+}
+
+void CSpeciesSolver::Preprocessing(CGeometry* geometry, CSolver** solver_container, CConfig* config,
+                                   unsigned short iMesh, unsigned short iRKStep, unsigned short RunTime_EqSystem,
+                                   bool Output) {
+  const bool implicit = (config->GetKind_TimeIntScheme() == EULER_IMPLICIT);
+  const bool muscl = config->GetMUSCL_Species();
+  const bool limiter =
+      (config->GetKind_SlopeLimit_Species() != NO_LIMITER) && (config->GetInnerIter() <= config->GetLimiterIter());
+
+  for (auto iPoint = 0u; iPoint < nPoint; iPoint++) {
```

Comment:
```suggestion
  SU2_OMP_FOR_STAT(omp_chunk_size)
  for (auto iPoint = 0ul; iPoint < nPoint; iPoint++) {
```

Reply:
Oh yes good point, I'll have to do a sweep through the code to check for loops that need to be made OMP ready 👍

Reply:
369b5b8 I think I captured all of em there, i hope I did them correctly